### DISCUSSION TITLE
Data segregation between Dev app and Published app doesn't seem to work?

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#3682] [Help]
I believe that the Dev app and the Prod apps run on completely different databases.
However, this doesn't seem to work correctly.

I am working on the `Job Application Tracker` sample app on a self-hosted version of Budibase `v 1.0.6.` ...
When I create an applicant from the public form, I can see the data in the dev version too!

This seems to be a bug? Pls advise if this is a bug?

Thank you!

**Comment by shogunpurple** [comment]
Hi @SantoshSrinivas79 - the dev version of the app syncs with the data in production. This ensures data integrity and avoids conflicts - if you change the same row in both dev and production, for example.

Syncing happens whenever you open the app in the builder.

**Reply by SantoshSrinivas79** [reply]
Thanks @shogunpurple! Just curious, why do dev and production need to sync at all?

In other words, I am thinking that the dev team should be able to work with *toy data* without necessarily having production-level data visibility or knowledge at all.

**Reply by shogunpurple** [reply]
As above - we do not want the dev/production application to have differing changes on the same record, as we do not have anything in place to resolve conflicts. I think what you are referring to, and what has been asked before is the ability to turn this off, and only sync the structure of your app and the schema, not the rows of data within the app itself. This would let you have 2 versions of the same app with different data sets.

**Reply by SantoshSrinivas79** [reply]
@shogunpurple I guess this makes sense.

Is this possible currently?

> I think what you are referring to, and what has been asked before is the ability to turn this off, and only sync the structure of your app and the schema, not the rows of data within the app itself. This would let you have 2 versions of the same app with different data sets.

**Reply by aderici** [reply]
If you have an app, already published, being used by actual users, how are you going to continue developing and testing this app, without hurting your production data? I might be missing something here but i created some test data in dev and they all synced to production when i published my app. and then i deleted all rows in dev and then when i published all my production data was gone. 

Can you help me understand what's the intended workflow here? I would like to be able to continue development and testing while users are using the app in prod

**Comment by CrashDown35** [comment]
I agree with @aderici and I am also facing the same issue.
Is the environment management that is in the roadmap going to address this?
